src/trans-word-level/property.cpp

```diff
@@ -108,11 +112,51 @@ static void property_obligations_rec(
         PRECONDITION(false);
     }(property_expr);
 
-    for(std::size_t c = current; c < no_timeframes; c++)
+    for(mp_integer c = current; c < no_timeframes; ++c)
     {
       property_obligations_rec(phi, solver, c, no_timeframes, ns, obligations);
     }
   }
+  else if(
+    property_expr.id() == ID_sva_ranged_always ||
+    property_expr.id() == ID_sva_s_always)
+  {
+    auto &phi = property_expr.id() == ID_sva_ranged_always
+                  ? to_sva_ranged_always_expr(property_expr).op()
+                  : to_sva_s_always_expr(property_expr).op();
+    auto &range = property_expr.id() == ID_sva_ranged_always
+                    ? to_sva_ranged_always_expr(property_expr).range()
+                    : to_sva_s_always_expr(property_expr).range();
+
+    mp_integer from, to;
+
+    auto from_opt = numeric_cast<mp_integer>(range.op0());
+    if(!from_opt.has_value())
+      throw ebmc_errort() << "failed to convert SVA always from index";
+
+    from = *from_opt;
+
+    if(range.op1().id() == ID_infinity)
+    {
+      to = no_timeframes - 1;
+    }
+    else
+    {
+      auto to_opt = numeric_cast<mp_integer>(range.op1());
+      if(!to_opt.has_value())
+        throw ebmc_errort() << "failed to convert SVA always to index";
+      to = *to_opt;
+    }
+
+    for(mp_integer c = from; c <= to; ++c)
+    {
+      if(c >= 0 && c < no_timeframes)
```

Comment:
A couple of (related) ideas:
* No need to declare `from` before it can be initialised.
* Taking this one step further, how about: `mp_integer from = std::max(0, *from_opt);`
* Use `to = std::min(no_timeframes - 1, *to_opt);`
These together will avoid unnecessary increments of `c` and permit removing the `if` inside this `for` loop.

Reply:
done